### PR TITLE
fix(dialog-full-screen): close button styling

### PR DESCRIPTION
**Pull request by damianOfKyan**
### Proposed behaviour
Update the DialogFullScreen close icon, when using a modern theme the icon will have a gold outline.

Screenshot 2020-03-04 at 11 22 11

### Current behaviour
The icon has a blue outline that is also mis-aligned.
Screenshot 2020-03-04 at 11 14 32

### Checklist

- [x] Release notes (Conventional Commits) 
- [x] Unit tests
- [x] Cypress automation tests
<del>- [x] Storybook added or updated
<del>- [x] Theme support
<del>- [ ] Typescript `d.ts` file added or updated

### Additional context
There are other outstanding PR's related to close icon.
#2690 #2692 #2693 #2694 

### Testing instructions
1. `npm start`
1. Go to http://localhost:9001/?path=/story/dialog-full-screen--default

### CODE REVIEW

File: src/components/dialog-full-screen/dialog-full-screen.component.js [hunk 1]
```diff
@@ -94,11 +97,17 @@ class DialogFullScreen extends Modal {
     return this.document.documentElement;
   }
 
-  onButtonKeyDown = (ev) => {
-    if (Events.isEnterKey(ev) || Events.isSpaceKey(ev)) {
-      ev.preventDefault();
-      this.props.onCancel();
-    }
+  get closeIcon() {
```
Comment (dawidzarzycki) [inline src/components/dialog-full-screen/dialog-full-screen.component.js]: Can you remove `get` please?

File: src/components/dialog-full-screen/dialog-full-screen.component.js [hunk 2]
```diff
@@ -27,7 +28,9 @@ class DialogFullScreen extends Modal {
 
   static defaultProps = {
     open: false,
-    enableBackgroundUI: true
+    enableBackgroundUI: true,
+    onCancel: null,
+    showCloseIcon: true
   }
```
Comment (dawidzarzycki) [inline src/components/dialog-full-screen/dialog-full-screen.component.js]: I suggest to replace this static with:
```
DialogFullScreen.defaultProps = {...}
```
to be consistent

File: src/components/dialog-full-screen/dialog-full-screen.component.js (hunk 1 above)
Comment (damianOfKyan) [reply src/components/dialog-full-screen/dialog-full-screen.component.js]: fixed: 2a1d3f6

File: src/components/dialog-full-screen/dialog-full-screen.component.js (hunk 2 above)
Comment (damianOfKyan) [reply src/components/dialog-full-screen/dialog-full-screen.component.js]: fixed: 2a1d3f6

File: cypress/features/regression/dialogFullScreen.feature [hunk 3]
```diff
@@ -4,6 +4,13 @@ Feature: Dialog Full Screen component
   Background: Open Dialog Full Screen component page
     Given I open "Dialog Full Screen" component page
 
+  @positive
+  Scenario: Clicking close icon closes Dialog Full Screen
+    When I check showCloseIcon checkbox
+      And I open component preview
+      And I click close icon
+    Then Confirm dialog is not visible
```
Comment (DlgSHi) [inline cypress/features/regression/dialogFullScreen.feature]:
```
Scenario: Clicking close icon closes Dialog Full Screen
     Given I check showCloseIcon checkbox
       And I open component preview
     When I click close icon
     Then Confirm dialog is not visible

File: cypress/features/regression/dialogFullScreen.feature [hunk 4]
```diff
@@ -4,6 +4,13 @@ Feature: Dialog Full Screen component
   Background: Open Dialog Full Screen component page
```
Comment (DlgSHi) [inline cypress/features/regression/dialogFullScreen.feature]: what about test for golden color?

Comment (damianOfKyan) [reply cypress/features/regression/dialogFullScreen.feature]: added: f6bc4b3

File: cypress/features/regression/dialogFullScreen.feature (hunk 3 above)
Comment (damianOfKyan) [reply cypress/features/regression/dialogFullScreen.feature]: fixed: f6bc4b3

File: src/components/dialog-full-screen/dialog-full-screen.component.js [hunk 5]
```diff
@@ -120,19 +122,29 @@ class DialogFullScreen extends Modal {
...
         { title }
       </FullScreenHeading>
     );
   }
 }
 
+DialogFullScreen.defaultProps = {
+  open: false,
+  enableBackgroundUI: true,
+  onCancel: null,
```
Comment (jamime) [inline src/components/dialog-full-screen/dialog-full-screen.component.js]:
```suggestion
```

File: src/components/dialog-full-screen/dialog-full-screen.spec.js [hunk 6]
```diff
@@ -44,6 +43,37 @@ describe('DialogFullScreen', () => {
...
+        </DialogFullScreen>
+      );
+      instance = wrapper.instance();
+    });
+
+    it('renders the dialog', () => {
+      expect(instance._dialog).toBeTruthy();
+    });
+
+    it('closes when the exit icon is click', () => {
```
Comment (jamime) [inline src/components/dialog-full-screen/dialog-full-screen.spec.js]: We should add the same tests that we added to `Dialog` (ESC, Enter, a)

Comment (damianOfKyan) [reply src/components/dialog-full-screen/dialog-full-screen.spec.js]: done.

File: src/components/dialog-full-screen/dialog-full-screen.component.js (hunk 5 above)
Comment (damianOfKyan) [reply src/components/dialog-full-screen/dialog-full-screen.component.js]: done.

**Pull request merged**